app/assets/javascripts/discourse/widgets/quick-access-bookmarks.js.es6

```diff
@@ -0,0 +1,59 @@
+import { h } from "virtual-dom";
+import DiscourseURL from "discourse/lib/url";
+import QuickAccessPanel from "discourse/widgets/quick-access-panel";
+import UserAction from "discourse/models/user-action";
+import { ajax } from "discourse/lib/ajax";
+import { createWidgetFrom } from "discourse/widgets/widget";
+import { postUrl } from "discourse/lib/utilities";
+
+const ICON = "bookmark";
+
+let staleItems = [];
+
+// The empty state help text for bookmarks page is localized on the server.
+let emptyStatePlaceholderItemText = "";
+
+createWidgetFrom(QuickAccessPanel, "quick-access-bookmarks", {
+  buildKey: () => "quick-access-bookmarks",
+
+  hasMore() {
+    // Always show the button to the bookmarks page.
+    return true;
+  },
+
+  showAll() {
+    DiscourseURL.routeTo(`${this.attrs.path}/activity/bookmarks`);
+  },
+
+  emptyStatePlaceholderItem() {
+    return h("li.read", emptyStatePlaceholderItemText);
+  },
+
+  findStaleItems() {
+    return staleItems || [];
+  },
+
+  findNewItems() {
+    return ajax("/user_actions.json", {
+      cache: "false",
+      data: {
+        username: this.currentUser.username,
+        filter: UserAction.TYPES.bookmarks,
+        limit: this.estimateItemLimit(),
+        no_results_help_key: "user_activity.no_bookmarks"
+      }
+    }).then(({ user_actions, no_results_help }) => {
+      emptyStatePlaceholderItemText = no_results_help;
+      return (staleItems = user_actions.slice(0, this.estimateItemLimit()));
```

Comment:
The `user_actions` endpoint does not support a `limit` parameter, so this truncates the results to ensure they will fit on the screen. I am not sure if it is preferred to add a new `limit` param to the endpoint?

Reply:
It  looks like it would be quite simple to add a `limit` parameter to the route, so I would suggest doing that.